internal/pkg/service/appsproxy/proxy/server_test.go

```diff
@@ -0,0 +1,204 @@
+	// Start app
+	appServer := testutil.StartAppServer(t)
+	defer appServer.Close()
+
+	// Start DNS server
+	dnsServer := testutil.StartDNSServer(t)
+	defer func() {
+		assert.NoError(t, dnsServer.Shutdown())
+	}()
+
+	// Register app DNS record
+	appURL := testutil.AddAppDNSRecord(t, appServer, dnsServer)
+
+	// Start api
+	appsAPI := testutil.StartDataAppsAPI(t, []api.AppConfig{
+		{
+			ID:             "123",
+			Name:           "public",
+			UpstreamAppURL: appURL.String(),
+			AuthRules: []api.Rule{
+				{
+					Type:         api.RulePathPrefix,
+					Value:        "/",
+					AuthRequired: pointer(false),
+				},
+			},
+		},
+	})
+	defer appsAPI.Close()
+
+	// Configure proxy
+	cfg := config.New()
+	cfg.DNSServer = dnsServer.Addr()
+	cfg.API.PublicURL, _ = url.Parse("https://hub.keboola.local")
+	cfg.SandboxesAPI.URL = appsAPI.URL
+
+	d, mocked := proxyDependencies.NewMockedServiceScope(t, cfg, dependencies.WithRealHTTPClient())
+
+	handler := proxy.NewHandler(d)
+
+	// Get robots.txt
+	rec := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "https://hub.keboola.local/robots.txt", nil)
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusOK, rec.Code)
+	assert.Contains(t, rec.Body.String(), "Disallow: /")
+
+	// Get style.css
+	rec = httptest.NewRecorder()
+	req = httptest.NewRequest("GET", "https://hub.keboola.local/_proxy/assets/style.css", nil)
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusOK, rec.Code)
+	assert.NotEmpty(t, rec.Body.String())
+
+	// Get missing asset
+	rec = httptest.NewRecorder()
+	req = httptest.NewRequest("GET", "https://hub.keboola.local/_proxy/assets/foo.bar", nil)
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusNotFound, rec.Code)
+
+	// Invalid host
+	rec = httptest.NewRecorder()
+	req = httptest.NewRequest("GET", "https://123.foo.bar.local/path", nil)
+	req.Header.Set("User-Agent", "my-user-agent")
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusBadRequest, rec.Code)
+	assert.Contains(t, rec.Body.String(), "Unexpected domain, missing application ID.")
+
+	// Send logged request
+	rec = httptest.NewRecorder()
+	req = httptest.NewRequest("GET", "https://123.hub.keboola.local/path", nil)
+	req.Header.Set("User-Agent", "my-user-agent")
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusOK, rec.Code)
+	assert.Equal(t, "Hello, client", rec.Body.String())
+
+	// Send ignored request
+	rec = httptest.NewRecorder()
+	req = httptest.NewRequest("GET", "https://hub.keboola.local/health-check", nil)
+	req.Header.Set("User-Agent", "my-user-agent")
+	handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusOK, rec.Code)
+	assert.Equal(t, "OK\n", rec.Body.String())
+
+	assert.NotNil(t, mocked.DebugLogger().CompareJSONMessages(`{"level":"info","message":"req https://123.hub.keboola.local/health-check %A"}`))
+	mocked.DebugLogger().AssertJSONMessages(t, `{"level":"info","message":"req https://123.hub.keboola.local/path %A","http.request_id":"%s","component":"http"}`)
```

Comment:
This looks like some table tests, 5 test cases, similar code, creating recorder request etc. I dont say that you need to do it now, but It would be fine if it is refactored as table tests. It also prevents writing commentaries which are part of table test `ID`.